lib/pages/login/login_page.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/material.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+import 'package:morning_weakers/pages/all/all_page.dart';
+
+class LoginPage extends StatefulWidget {
+  const LoginPage({Key key, this.title}) : super(key: key);
+  final String title;
+
+  @override
+  _MyHomePageState createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<LoginPage> {
+  final GoogleSignIn _googleSignIn = GoogleSignIn();
+  final FirebaseAuth _auth = FirebaseAuth.instance;
+
+  Future<FirebaseUser> _handleSignIn() async {
+    GoogleSignInAccount googleCurrentUser = _googleSignIn.currentUser;
+    try {
+      googleCurrentUser ??= await _googleSignIn.signInSilently();
+      googleCurrentUser ??= await _googleSignIn.signIn();
+      if (googleCurrentUser == null) return null;
+
+      final GoogleSignInAuthentication googleAuth = await googleCurrentUser.authentication;
+      final AuthCredential credential = GoogleAuthProvider.getCredential(
+        accessToken: googleAuth.accessToken,
+        idToken: googleAuth.idToken,
+      );
+      final FirebaseUser user = (await _auth.signInWithCredential(credential)).user;
+      print('signed in ${user.displayName}');
+
+      return user;
+    }
+    on Exception catch (error) {
+      print(error);
+      return null;
+    }
+  }
+
+  void transitionNextPage(FirebaseUser user) {
+    if (user == null) return;
+
+    Navigator.push(context, MaterialPageRoute(builder: (context) =>
+        NextPage(userData: user)
+    ));
```

Comment:
```suggestion
    Navigator.push<void>(context, MaterialPageRoute(builder: (context) =>
        NextPage(userData: user)
    ));
```
lint的にpushのあとに型を買いた方が良さげ！